### PR TITLE
Local tests setup

### PR DESCRIPTION
**Pull request by navjot50**
Automated the setup of local Wordpress instance to get a totally ready container to run integration tests against.

A python script makes use of selenium to interact with the Wordpress instance and completes the installation of Wordpress, sets up the Wordpress user, sets up Wordpress settings, configures and activates all the required plugins.

At the moment the enable-application-passwords plugin is added in the dev folder in tar.gz compression, however this can be totally removed from source of this repo if you approve the PR https://github.com/wp-net/wordpress-docker-compose/pull/1 , then the plugin can be directly taken from the web.

Finally, containers-startup scripts are added with sh and ps1 extension, which are the startup scripts for all the required containers.

### CODE REVIEW

**Comment by ThomasPe** [comment]
wow, a lot to unpack - thank you so much for the effort. 
I haven't worked with selenium myself yet (I understand what it does) - is there any advantage doing the setup like this instead of using wpcli like we do here? https://github.com/wp-net/wordpress-docker-compose

**Comment by navjot50** [comment]
You are welcome. Personally, I never knew about wp-cli, which led me to think of selenium to do some UI interactions to setup the wordpress instance. Selenium had a nice support for docker as well. However, I will have a look at the wp-cli way that your link suggests. If you think the wp-cli way is easier, then I may contribute via a new branch with wp-cli automation.

**Comment by ThomasPe** [comment]
I don't know about easier, that's why I haven't set it up yet for local debugging. :D 
I see your script basically just does the manual steps described in install.md, let me run it to see how it works for me 👍

File: dev/containers-startup.ps1 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+docker-compose rm -f
+docker-compose build
+docker-compose up --no-build -d
+docker build -t script ./wordpressbot/
+docker run -it --network=integration-tests-setup --name=script script
+docker stop selenium && docker rm selenium
```
[the remark this answers is not in the document]
Comment (navjot50) [reply dev/containers-startup.ps1:6]: Ok. I will do it. I was on a Mac so I couldn't test it first hand on a Windows.

**Comment by ThomasPe** [comment]
I'm still running into a few issues on windows, mainly because using port 80 is blocked by (I assume) IIS. When changing the port to 8080 the http://wordpress doesn't seem to work anymore, and with localhost:8080 or 127.0.0.1:8080 the selenium container does not have access to WordPress it seems.

**Comment by navjot50** [comment]
Ok. Yes, if you change the port to 8080 then the selenium we will need to change the uri in the selenium script. I went ahead with 80 because in the `WordPressPCL.Tests.Selfhosted.Utility.ApiCredentials` we have the property `public static string WordPressUri = "http://localhost/wp-json/";` which is taking port number 80. I can change everything to 8080 but this will require to change `WordpressUri` to `http://localhost:8080/wp-json/`

**Comment by ThomasPe** [comment]
That would be great, thanks

**Comment by navjot50** [comment]
Changed the Wordpress port to 8080 for host machine and changed the script slightly to work for all powershell versions and not just for powershell 7 and above.

**Comment by navjot50** [comment]
Changed the Wordpress port to 8080 for host machine and changed the script slightly to work for all powershell versions and not just for powershell 7 and above. Also, now the enable-application-passwords plugin is being fetched from web and it is removed from the source. I have tested it on Mac. I will be further testing it on a Windows tomorrow but I am hopeful that it should work now (unless docker behaves differently on Mac and Windows - which ideally it should not).

**Comment by navjot50** [comment]
Yes, the wordpress is not accessible via browser on the local (for that a reverse proxy like nginx will be needed). However, that is not needed because the purpose of this container is to only run tests against. I was able to run the test suite both in Jetbrains Rider and VS2019 once I have the containers script fully executed. What error did it give you while running the self hosted test suite?

**Pull request merged**